src/language-server/project/rover/project.ts

```diff
@@ -108,30 +162,40 @@ export class RoverProject extends GraphQLProject {
       console.error({ err });
     });
 
+    connection.onNotification(
+      PublishDiagnosticsNotification.type,
+      this.documents.handlePartDiagnostics.bind(this.documents),
+    );
+
+    connection.onUnhandledNotification((notification) => {
+      DEBUG && console.info("unhandled notification from LSP", notification);
+    });
+
     connection.listen();
-    console.log("Initializing connection");
+    DEBUG && console.log("Initializing connection");
 
     const source = new CancellationTokenSource();
-    return connection
-      .sendRequest(
+    try {
+      const status = await connection.sendRequest(
         InitializeRequest.type,
         {
           capabilities: this.capabilities,
           processId: process.pid,
           rootUri: this.rootURI.toString(),
         },
         source.token,
-      )
-      .then(
-        (status) => {
-          console.log("Connection initialized", status);
-          return connection;
-        },
-        (error) => {
-          console.error("Connection failed to initialize", error);
-          throw error;
-        },
       );
+      DEBUG && console.log("Connection initialized", status);
+
+      await this.connectionStorage.run(connection, () =>
+        this.documents.resendAllDocuments(),
+      );
```

Comment:
```suggestion
      await this.connectionStorage.run(connection, this.documents.resendAllDocuments.bind(this.documents);
```

Since you're wanting to avoid closure-related memory leaks 🙂